x-pack/plugins/security_solution/public/common/components/url_state/use_url_state.tsx

```diff
@@ -115,18 +108,12 @@ export const useUrlStateHooks = ({
         urlState[urlKey] != null &&
         urlState[urlKey]?.query === ''
       ) {
-        mySearch = replaceStateInLocation({
-          history,
-          pathName,
-          search: mySearch,
+        statesToUpdate.push({
```

Comment:
Add the state to update to an array instead of updating it many times.
Before we would call `history.replace` for every key in the query string.